container/check_container.go

```diff
@@ -55,44 +86,30 @@ func (c *containerCheck) Run(ctx context.Context) (certification.Results, error)
 
 		override, err := lib.GetContainerPolicyExceptions(ctx, p)
 		if err != nil {
-			return certification.Results{}, fmt.Errorf("%w: %s", preflighterr.ErrCannotResolvePolicyException, err)
+			return fmt.Errorf("%w: %s", preflighterr.ErrCannotResolvePolicyException, err)
 		}
 
-		pol = override
+		c.policy = override
 	}
-
-	checks, err := engine.InitializeContainerChecks(ctx, pol, engine.ContainerCheckConfig{
+	var err error
+	c.checks, err = engine.InitializeContainerChecks(ctx, c.policy, engine.ContainerCheckConfig{
```

Comment:
I'm not a fan of updating fields directly like this, especially when there is a possible error involved. The field should only be updated one it has been determined that there is no error.

Reply:
fixed